File: backend/src/routes/habits.routes.ts
import { Router } from 'express'
import { requireAuth } from '../middlewares/auth.middleware'
import {
  listarHabitos,
  criarHabito,
  editarHabito,
  deletarHabito,
  toggleConclusao,
} from '../controllers/habits.controller'

const router = Router()

// Todas as rotas de hábitos exigem usuário autenticado
router.use(requireAuth)

// GET  /api/habits          → lista hábitos com conclusões e sequência
// POST /api/habits          → cria novo hábito
router.get('/',  listarHabitos)
router.post('/', criarHabito)

// PUT    /api/habits/:id    → edita nome, ícone ou cor
// DELETE /api/habits/:id    → remove hábito e suas conclusões
router.put('/:id',    editarHabito)
router.delete('/:id', deletarHabito)

// POST /api/habits/:id/toggle   → marca/desmarca uma data  { date: "YYYY-MM-DD" }
router.post('/:id/toggle', toggleConclusao)

export default router

File: backend-go/internal/habits/habits_routes.go
package habits

import (
	"github.com/gin-gonic/gin"
	"github.com/jackc/pgx/v5/pgxpool"

	"beeplanerofc/backend-go/internal/middleware"
	"beeplanerofc/backend-go/internal/supabaseauth"
)

// RegisterRoutes registra as rotas de hábitos no grupo /api/habits. Todas exigem login.
func RegisterRoutes(router *gin.RouterGroup, db *pgxpool.Pool, authClient *supabaseauth.Client) {
	ctrl := NewController(db)

	router.Use(middleware.RequireAuth(db, authClient))

	router.GET("", ctrl.ListarHabitos)
	router.POST("", ctrl.CriarHabito)
	router.PUT("/:id", ctrl.EditarHabito)
	router.DELETE("/:id", ctrl.DeletarHabito)
	router.POST("/:id/toggle", ctrl.ToggleConclusao)
}
